Core/Core/CourseSync/CourseSyncProgress/Model/CourseSyncProgressInteractor.swift

```diff
@@ -0,0 +1,87 @@
+//
+// This file is part of Canvas.
+// Copyright (C) 2023-present  Instructure, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import Combine
+import CombineExt
+import Foundation
+
+struct SyncProgress {
+    let total: Int64
+    let progress: Int64
+    let failure: Bool
+}
+
+protocol CourseSyncProgressInteractor {
+    func getSyncProgress() -> SyncProgress
+    func getCourseSyncProgressEntries() -> AnyPublisher<[CourseSyncEntry], Error>
```

Comment:
I know this is a temporary interactor but I'd still unify these methods and make them reactive so it's less work later on.